src/offspot_demo/deploy.py

```diff
@@ -1,2 +1,345 @@
+""" Deploy an image from an image URL
+
+Limitations:
+- URL must be hosted on S3 (or at least webserver must serve an md5 digest in ETag)
+- Image must be Imager-service-created: have its data in a 3rd ext4 partition
+"""
+
+import argparse
+import hashlib
+import http
+import logging
+import subprocess
+import sys
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import NamedTuple
+
+import requests
+
+from offspot_demo.__about__ import __version__
+from offspot_demo.constants import (
+    DOCKER_LABEL_MAINT,
+    IMAGE_PATH,
+    TARGET_DIR,
+    Mode,
+    get_logger,
+)
+from offspot_demo.prepare import prepare_image
+from offspot_demo.toggle import toggle_demo
+from offspot_demo.utils.image import (
+    attach_to_device,
+    detach_device,
+    get_loopdev,
+    get_loopdev_used_by,
+    is_mounted,
+    mount_on,
+    unmount,
+)
+
+ONE_MIB = 2**20
+
+logger = get_logger("deploy")
```

Comment:
I recommend to move this to `constants.py` and only import logger, so that we do not finish with various logger names / log levels in various modules of same project (already seen in scrapers ^^)

Reply:
I agree ; will be main entrypoint (watcher)'s responsibility to update the log level then